query-engine/driver-adapters/src/conversion.rs

```diff
@@ -35,6 +37,23 @@ impl ToNapiValue for JSArg {
             JSArg::Buffer(bytes) => {
                 ToNapiValue::to_napi_value(env, napi::Env::from_raw(env).create_buffer_with_data(bytes)?.into_raw())
             }
+            // While arrays are encodable as JSON generally, their element might not be, or may be
+            // represented in a different way than we need. We use this custom logic for all arrays
+            // to avoid having separate `JsonArray` and `BytesArray` variants in `JSArg` and
+            // avoid complicating the logic in `conv_params`.
+            JSArg::Array(items) => {
+                let env = napi::Env::from_raw(env);
+                let mut array = env.create_array(items.len().try_into().expect("JS array length must fit into u32"))?;
+
+                for (index, item) in items.into_iter().enumerate() {
+                    let js_value = ToNapiValue::to_napi_value(env.raw(), item)?;
+                    // TODO: NapiRaw could be implemented for sys::napi_value directly, there should
+                    // be no need for re-wrapping; submit a patch to napi-rs and simplify here.
+                    array.set(index as u32, napi::JsUnknown::from_raw_unchecked(env.raw(), js_value))?;
```

Comment:
Note: `from_raw_unchecked` is safe here for two reasons:
1. It's `JsUnknown` anyway
2. Even if the type didn't match it still wouldn't matter as we never actually use it as a high-level `NapiValue`, its whole purpose is to be immediately destructured back into a raw `sys::napi_value`.

We only use it here as a trick because `napi::sys::napi_value` itself doesn't currently implement `NapiRaw`, we could also wrap it into our own newtype for the same effect to implement it ourselves without violating coherence/orphan rules:

```rust
struct JsValue(napi::sys::napi_value);

impl napi::NapiRaw for JsValue {
  unsafe fn raw(&self) -> napi::sys::napi_value {
    self.0
  }
}
```

but wrapping and unwrapping with `JsUnknown` is good enough as it should be pretty much zero cost with `from_raw_unchecked`, maybe modulo some extra bytes on the stack.